libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/RouteViewModel.java

```diff
@@ -157,6 +159,8 @@ private boolean launchWithRoute(NavigationViewOptions options) {
   private void extractRoute(NavigationViewOptions options) {
     DirectionsRoute route = options.directionsRoute();
     if (route != null) {
+      String profile = options.directionsProfile();
+      routeProfile = profile != null ? profile : route.routeOptions().profile();
```

Comment:
Several checks are already in place to ensure this is never null including the [MAS Directions builder](https://github.com/mapbox/mapbox-java/blob/master/services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java#L331) and the server response itself. I've opened https://github.com/mapbox/mapbox-java/issues/683 to change the annotation for this to a NonNull

Reply:
@cammace awesome, thanks for the heads up on this

Reply:
Hey, it is posible to pass Locale when reroute? seems to be the answer for issue #633

Reply:
@darkwizzard PR is waiting for review https://github.com/mapbox/mapbox-navigation-android/pull/635